File: LSL/liblsl-Python/examples/SendStringMarkers.py
"""Example program to demonstrate how to send string-valued markers into LSL."""

import random
import time

from pylsl import StreamInfo, StreamOutlet

# first create a new stream info (here we set the name to MyMarkerStream,
# the content-type to Markers, 1 channel, irregular sampling rate,
# and string-valued data) The last value would be the locally unique
# identifier for the stream as far as available, e.g.
# program-scriptname-subjectnumber (you could also omit it but interrupted
# connections wouldn't auto-recover). The important part is that the
# content-type is set to 'Markers', because then other programs will know how
#  to interpret the content
info = StreamInfo('MyMarkerStream', 'Markers', 1, 0, 'string', 'myuidw43536')

# next make an outlet
outlet = StreamOutlet(info)

print("now sending markers...")
markernames = ['Test', 'Blah', 'Marker', 'XXX', 'Testtest', 'Test-1-2-3']
while True:
    # pick a sample to send an wait for a bit
    outlet.push_sample([random.choice(markernames)])
    time.sleep(random.random()*3)

File: LSL/liblsl-Java/examples/SendStringMarkers.java
package examples;
import edu.ucsd.sccn.LSL;
import java.io.IOException;

public class SendStringMarkers {
    public static void main(String[] args) throws IOException, InterruptedException  {
        System.out.println("Creating a new StreamInfo...");
        LSL.StreamInfo info = new LSL.StreamInfo("MyMarkers","Markers",1,LSL.IRREGULAR_RATE,LSL.ChannelFormat.string,"myuid4563");

        System.out.println("Creating an outlet...");
        LSL.StreamOutlet outlet = new LSL.StreamOutlet(info);
        
        System.out.println("Sending data...");
        String[] strings = {"Test 1-2-3","Blah","ABC"};
        String[] sample = new String[1];
        for (int t=0;t<100000;t++) {
            sample[0] = strings[t%3];
            outlet.push_sample(sample);
            Thread.sleep((long)(Math.random()*2000));
        }
        
        outlet.close();
        info.destroy();
    }
}